postgrest.cabal

```diff
@@ -38,7 +38,9 @@ library
   exposed-modules:    PostgREST.ApiRequest
                       PostgREST.App
                       PostgREST.Auth
+                      PostgREST.CLI
                       PostgREST.Config
+                      PostgREST.Connection
```

Comment:
I think it was mainly to allow Haskell devs to use postgrest as a lib and extend it. Not that I have ever done that, but I'm not sure if we're invalidating haskell libs such as https://github.com/diogob/postgres-websockets

Reply:
We need both the lib and executable: We need modules to be in a lib in order to test them, and we obviously need an executable to generate `postgrest`. What I see most projects do is to pull as much as possible into the lib (where it can be tested and inspected) and have a minimal one-liner Main module. E.g. Pandoc: https://github.com/jgm/pandoc/blob/master/app/pandoc.hs

In our case, we have the extra wart of having to use CPP macros to conditionally compile in non-Windows features. it's good to isolate those in Main I think, while having as much as possible available for testing in the lib.

Reply:
> I think it was mainly to allow Haskell devs to use postgrest as a lib and extend it. Not that I have ever done that, but I'm not sure if we're invalidating haskell libs such as https://github.com/diogob/postgres-websockets

Doesn't look like anyone is using PostgREST as a lib:
https://packdeps.haskellers.com/reverse